craft_application/services/provider.py

```diff
@@ -206,12 +206,12 @@ def get_provider(self, name: str | None = None) -> craft_providers.Provider:
             emit.debug(f"Using provider {name!r} passed as an argument.")
             chosen_provider: str = name
 
-        # (2) get the provider from the environment (CRAFT_BUILD_ENVIRONMENT),
-        elif env_provider := os.getenv("CRAFT_BUILD_ENVIRONMENT"):
-            emit.debug(f"Using provider {env_provider!r} from environment.")
-            chosen_provider = env_provider
+        # (2) get the provider from build_environment
+        elif provider := self._services.config.get("build_environment"):
+            emit.debug(f"Using provider {provider!r} from system configuration.")
+            chosen_provider = provider
 
-        # (3) use provider specified with snap configuration,
+        # (3) use provider specified in provider
```

Comment:
Seems a little weirdly phrased
```suggestion
        # (3) use provider specified in snap configuration
```